presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -411,18 +416,11 @@ public synchronized void updatePartitionStatistics(HiveIdentity identity, Table
     @Override
     public synchronized List<String> getAllTables(String databaseName)
     {
-        requireNonNull(databaseName, "databaseName is null");
-
-        Optional<Database> database = getDatabase(databaseName);
-        if (database.isEmpty()) {
-            return ImmutableList.of();
-        }
-
-        Path databaseMetadataDirectory = getDatabaseMetadataDirectory(databaseName);
-        List<String> tables = getChildSchemaDirectories(databaseMetadataDirectory).stream()
-                .map(Path::getName)
+        return listAllTables(databaseName).stream()
+                .filter(hideDeltaLakeTables
+                        ? Predicate.not(ImmutableSet.copyOf(getTablesWithParameter(databaseName, SPARK_TABLE_PROVIDER_KEY, DELTA_LAKE_PROVIDER))::contains)
```

Comment:
static import.
Though I personally would prefer not using `Predicate.not` at all and just extend the lambda, and use `!`

Reply:
static import what?

> Though I personally would prefer not using Predicate.not at all and just extend the lambda, and use !

the `not` is necessary to make `ImmutableSet.copyOf(....)` computed once